rest-api/dao/participant_counts_over_time_service.py

```diff
@@ -176,70 +192,163 @@ def get_enrollment_status_sql(self, filters_sql_ps, filters_sql_p):
-                                    COALESCE(sample_order_status_2ed10_time, '3000-01-01'),
-                                    COALESCE(sample_order_status_1ed10_time, '3000-01-01'),
-                                    COALESCE(sample_order_status_1sal_time, '3000-01-01'),
-                                    COALESCE(sample_order_status_1sal2_time, '3000-01-01')
-                                    )
-                           END)
-                END) day
-               FROM participant_summary ps
-              %(filters_ps)s
-           GROUP BY day) full
-             ON c2.day = full.day) day_sums, calendar
-          WHERE calendar.day >= :start_date
-            AND calendar.day <= :end_date
-          GROUP BY calendar.day
-          ORDER BY calendar.day;
-      """ % {'filters_ps': filters_sql_ps, 'filters_p': filters_sql_p}
+                       LEFT OUTER JOIN participant_summary ps ON p.participant_id = ps.participant_id
+                %(filters_p)s) a left join
+               (select participant_id,
+                      (
+                          CASE
+                            WHEN enrollment_status>=2 and ps.consent_for_electronic_health_records = 1 THEN DATE(COALESCE(ps.consent_for_electronic_health_records_time, '3000-01-01'))
+                            ELSE Date('3000-01-01') END
+                          ) member_day,
+                       case when enrollment_status=3 then
+                       DATE(
+                         GREATEST(
+                           GREATEST(
+                             COALESCE(sign_up_time, '1000-01-01'),
+                             COALESCE(consent_for_electronic_health_records_time, '1000-01-01'),
+                             COALESCE(questionnaire_on_the_basics_time, '1000-01-01'),
+                             COALESCE(questionnaire_on_lifestyle_time, '1000-01-01'),
+                             COALESCE(questionnaire_on_overall_health_time, '1000-01-01'),
+                             COALESCE(physical_measurements_finalized_time, '1000-01-01')
+                               ),
+                           LEAST(
+                             COALESCE(sample_status_1ed04_time, '3000-01-01'),
+                             COALESCE(sample_status_2ed10_time, '3000-01-01'),
+                             COALESCE(sample_status_1ed10_time, '3000-01-01'),
+                             COALESCE(sample_status_1sal_time, '3000-01-01'),
+                             COALESCE(sample_status_1sal2_time, '3000-01-01')
+                               ))
+                       )
+                       else
+                           Date('3000-01-01')
+                       end core_day
+               from participant_summary ps
+               %(filters_ps)s) b
+               on b.participant_id=a.participant_id
+          where register_day<=:end_date or register_day='3000-01-01'
```

Comment:
Why is register_day = '3000-01-01' needed?

Reply:
I left join the register ones with member and core, so we need to keep all the participants, if remove this, there will be only register ones.